### PR TITLE
Make ProductArea badges interactive

### PR DESCRIPTION
**Pull request by jeffdaley**
Makes ProductArea badges clickable to initiate a product filter. Badges on the `/my` and `/drafts` are scoped to keep you on that route. All other users point to `/all`.

### CODE REVIEW

File: web/app/components/doc/tile.hbs [hunk 1]
```diff
@@ -1,44 +1,56 @@
...
+            {{@docNumber}}
+          </small>
+        {{/if}}
+      </div>
+      <div class="space-y-1 mt-2 max-w-full">
+        <Person
+          @ignoreUnknown={{true}}
+          @imgURL={{@avatar}}
+          @email="{{@owner}}"
+        />
```
Comment (jeffdaley) [inline web/app/components/doc/tile.hbs:42]: We can make this clickable in another PR.

File: web/app/components/doc/tile.hbs [hunk 2]
```diff
@@ -1,44 +1,56 @@
...
+  <LinkTo
+    @route="authenticated.document"
+    @model={{@docID}}
+    @query={{hash draft=@isDraft}}
+    class="pb-2 block"
+  >
+    {{!
+      We create a click area that extends beyond the edges of its relative container.
+      This makes the parent div clickable without having to wrap itself in a link,
+      and lets us nest interactive elements (e.g., ProductBadgeLink) in a way that improves the mouse experience without sacrificing accessibility.
```
Comment (jfreda) [inline web/app/components/doc/tile.hbs:11]: ❤️

File: web/app/components/product-badge-link.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+      return "authenticated.drafts";
+    }
+
+    if (currentRouteName.includes("my")) {
+      return "authenticated.my";
+    }
+
+    return "authenticated.all";
+  }
+}
```
Comment (jfreda) [inline web/app/components/product-badge-link.ts:54]: I couldn't immediately tell why we needed this logic - maybe a comment would help?

**Review by jfreda — APPROVED** [review]

File: web/app/components/product-badge-link.ts (hunk 3 above)
Comment (jeffdaley) [reply web/app/components/product-badge-link.ts]: Yeah I probably overcomplicated it anyway 😅

Updated
